editor/gui/editor_quick_open_dialog.cpp

```diff
@@ -538,35 +538,32 @@ void QuickOpenResultContainer::handle_search_box_input(const Ref<InputEvent> &p_
 	}
 
 	Ref<InputEventKey> key_event = p_ie;
-	if (key_event.is_valid() && key_event->is_pressed()) {
-		bool move_selection = false;
-
-		switch (key_event->get_keycode()) {
-			case Key::UP:
-			case Key::DOWN:
-			case Key::PAGEUP:
-			case Key::PAGEDOWN: {
-				move_selection = true;
-			} break;
-			case Key::LEFT:
-			case Key::RIGHT: {
-				if (content_display_mode == QuickOpenDisplayMode::GRID) {
-					// Maybe strip off the shift modifier to allow non-selecting navigation by character?
-					if (key_event->get_modifiers_mask() == 0) {
-						move_selection = true;
-					}
-				}
-			} break;
-			default:
-				break; // Let the event through so it will reach the search box.
-		}
+	if (key_event.is_null() || !key_event->is_pressed()) {
+		return;
+	}
 
-		if (move_selection) {
-			_move_selection_index(key_event->get_keycode());
-			queue_redraw();
-			accept_event();
+	Key move_selection = Key::NONE;
```

Comment:
`move_selection` can be a bool here, like before, since there is not really any benefit to save the `Key` here, unless I miss something?

Reply:
`_move_selection_index()` takes Key argument. It's simpler this way than rewriting its logic.

Reply:
I mean, the Key here is used as navigation enum, not actual key code. `_move_selection_index()` still somehow needs to decide what movement it should perform.